src/code/audio/audio_synthesis.c

```diff
@@ -1,137 +1,1700 @@
+            AudioSynth_ClearBuffer(cmd++, combFilterDmem, combFilterSize);
+            synthState->combFilterNeedsInit = false;
+        } else {
+            AudioSynth_LoadBuffer(cmd++, combFilterDmem, combFilterSize, combFilterState);
+        }
+        AudioSynth_SaveBuffer(cmd++, DMEM_TEMP + (numSamplesPerUpdate * SAMPLE_SIZE) - combFilterSize, combFilterSize,
+                              combFilterState);
+        AudioSynth_Mix(cmd++, (numSamplesPerUpdate * (s32)SAMPLE_SIZE) >> 4, combFilterGain, DMEM_COMB_TEMP,
+                       combFilterDmem);
+        AudioSynth_DMemMove(cmd++, combFilterDmem, DMEM_TEMP, numSamplesPerUpdate * SAMPLE_SIZE);
+    } else {
+        synthState->combFilterNeedsInit = true;
+    }
+
+    // Determine the behavior of the audio processing that leads to the haas effect
+    if ((sampleState->haasEffectLeftDelaySize != 0) || (synthState->prevHaasEffectLeftDelaySize != 0)) {
+        haasEffectDelaySide = HAAS_EFFECT_DELAY_LEFT;
+    } else if ((sampleState->haasEffectRightDelaySize != 0) || (synthState->prevHaasEffectRightDelaySize != 0)) {
+        haasEffectDelaySide = HAAS_EFFECT_DELAY_RIGHT;
+    } else {
+        haasEffectDelaySide = HAAS_EFFECT_DELAY_NONE;
+    }
+
+    // Apply an unknown effect based on the surround sound-mode
+    if (gAudioContext.soundMode == SOUNDMODE_SURROUND) {
+        sampleState->targetVolLeft = sampleState->targetVolLeft >> 1;
+        sampleState->targetVolRight = sampleState->targetVolRight >> 1;
+        if (sampleState->surroundEffectIndex != 0xFF) {
+            cmd = AudioSynth_ApplySurroundEffect(cmd, sampleState, synthState, numSamplesPerUpdate, DMEM_TEMP, flags);
+        }
+    }
+
+    // Split the mono-signal into left and right channels:
+    // Both for dry signal (to go to the speakers now)
+    // and for wet signal (to go to a reverb buffer to be stored, and brought back later to produce an echo)
+    cmd = AudioSynth_ProcessEnvelope(cmd, sampleState, synthState, numSamplesPerUpdate, DMEM_TEMP, haasEffectDelaySide,
+                                     flags);
+
+    // Apply the haas effect by delaying either the left or the right channel by a small amount
+    if (sampleState->bitField1.useHaasEffect) {
+        if (!(flags & A_INIT)) {
+            flags = A_CONTINUE;
+        }
+        cmd = AudioSynth_ApplyHaasEffect(cmd, sampleState, synthState, numSamplesPerUpdate * (s32)SAMPLE_SIZE, flags,
+                                         haasEffectDelaySide);
+    }
+
+    return cmd;
+}
+
+Acmd* AudioSynth_ApplySurroundEffect(Acmd* cmd, NoteSampleState* sampleState, NoteSynthesisState* synthState,
+                                     s32 numSamplesPerUpdate, s32 haasDmem, s32 flags) {
+    s32 wetGain;
+    u16 dryGain;
+    s64 dmem = DMEM_SURROUND_TEMP;
+    f32 decayGain;
+
+    AudioSynth_DMemMove(cmd++, haasDmem, DMEM_HAAS_TEMP, numSamplesPerUpdate * SAMPLE_SIZE);
+    dryGain = synthState->surroundEffectGain;
+
+    if (flags == 1) {
```

Comment:
Based on this it would seem like flags is a boolean? Looks like it is meant to clear the buffer.

Reply:
In this case, this flags has a define for it: `A_INIT`